c2corg_ui/static/js/documentediting.js

```diff
@@ -374,6 +390,25 @@ app.DocumentEditingController.prototype.handleMapFeatureChange_ = function(
 
 
 /**
+ * It doesn't associate, it just pushes it into associations array.
+ * @private
+ */
+app.DocumentEditingController.prototype.pushDocToAssociations_ = function() {
+  var doctype = Object.keys(this.urlParams_)[0];
+  var url = '/search?q=' + this.urlParams_[doctype] + '&t=' + doctype[0];
+  // get the doc from the API like /search?q=12345&t=r  (routes)
+  this.api_.getJson_(url).then(function(doc) {
```

Comment:
I think that is not what @gberaudo means. Create a function `searchDocumentById(document_id, doc_type)` which is called from here. The idea is that calls to the API are made with this service.